### PR TITLE
PROD-856Fixed-Issue with Apostrophe in the Name

### LINKED ISSUE #2139: Issue with Apostrophe in the Name
**Describe the bug**
Hi, Even though it works correctly throughout the site. But like ( O'Donoghue's ) apostrophe in her surname becomes glitchy in messages. 

**To Reproduce**

https://monosnap.com/file/6buhkJ6nN6O7hJNKUEUtVgbDGKEJuz

**Support ticket links**
https://secure.helpscout.net/conversation/1423108292/124706?folderId=3955985

**Jira issue** : [PROD-856]

[PROD-856]: https://buddyboss.atlassian.net/browse/PROD-856?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### PR DESCRIPTION
**Pull request by meetmakadiya**
### All Submissions:

* [x] Have you followed the [Pull Requests Guidelines](https://github.com/buddyboss/buddyboss-platform/wiki/Submitting-Pull-Requests#pull-request-guidelines)?
* [x] Does your code follow the [WordPress' Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../pulls) for the same update/change?

### Changes proposed in this Pull Request:

Fixes #2139 

### How to test the changes in this Pull Request:

1. update the user details with Apostrophe name
2. send a message from that account.
3. receiver person that time Apostrophe code on a name that issue fixed this PR

### Proof Screenshots or Video

http://i.imgur.com/YTAYqab.png

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [x] Have you written new tests for your changes, as applicable?
* [ ] Have you successfully run tests with your changes locally?

### Changelog entry

> Enter a summary of all changes on this Pull Request. This will appear in the changelog if accepted.

### CODE REVIEW

File: src/bp-templates/bp-nouveau/js/buddypress-messages.js [hunk 1]
```diff
@@ -626,7 +626,7 @@ window.bp = window.bp || {};
 						resp.messages[index].id                = value.id;
 						resp.messages[index].content           = value.content;
 						resp.messages[index].sender_id         = value.sender_id;
-						resp.messages[index].sender_name       = value.sender_name;
+						resp.messages[index].sender_name       = $( $.parseHTML( value.sender_name ) ).text();
```
Comment (dipskakadiya) [inline src/bp-templates/bp-nouveau/js/buddypress-messages.js:629]: @meetmakadiya It's nice to validate that at JS level but we should update this at the PHP level too.

**Review by dipskakadiya — CHANGES_REQUESTED** [review]
@meetmakadiya Please Check my comment.

File: src/bp-templates/bp-nouveau/js/buddypress-messages.js (hunk 1 above)
Comment (meetmakadiya) [reply src/bp-templates/bp-nouveau/js/buddypress-messages.js:629]: @dipskakadiya this value pass to the template so we do not need to validate on PHP side, please share your input and find the attachment.
http://i.imgur.com/G42go19.png

Comment (meetmakadiya) [reply src/bp-templates/bp-nouveau/js/buddypress-messages.js:629]: @dipskakadiya added validation on PHP side for sender_name

Comment (dipskakadiya) [reply src/bp-templates/bp-nouveau/js/buddypress-messages.js:629]: @meetmakadiya On frontend side `parseHTML` is not needed anymore

**Review by dipskakadiya — CHANGES_REQUESTED** [review]

**Comment by meetmakadiya** [comment]
@dipskakadiya revert the js file changes.

**Review by dipskakadiya — APPROVED** [review]

**Pull request closed without merging**